### ISSUE TITLE
Crash:  Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent

### ISSUE THREAD

**Issue by SimonIT** [ligi/PassAndroid#435]
Hi, after the recent update the whole app crashes at start:
```
04-18 18:35:05.758  8663  8663 D AndroidRuntime: Shutting down VM
04-18 18:35:05.759  8663  8663 E AndroidRuntime: FATAL EXCEPTION: main
04-18 18:35:05.759  8663  8663 E AndroidRuntime: Process: org.ligi.passandroid, PID: 8663
04-18 18:35:05.759  8663  8663 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{org.ligi.passandroid/org.ligi.passandroid.ui.PassListActivity}: org.koin.core.error.InstanceCreationException: Could not create instance for '[Singleton:'org.ligi.passandroid.Tracker']'
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3868)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4010)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:111)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2466)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:240)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:351)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:8380)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)04-18 18:35:05.759  8663  8663 E AndroidRuntime: Caused by: org.koin.core.error.InstanceCreationException: Could not create instance for '[Singleton:'org.ligi.passandroid.Tracker']'
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.koin.core.instance.InstanceFactory.a(SourceFile:147)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.koin.core.instance.SingleInstanceFactory.a(SourceFile:10)04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.koin.core.instance.SingleInstanceFactory$get$1.a(SourceFile:15)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.koin.core.instance.SingleInstanceFactory$get$1.invoke(Unknown Source:0)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.koin.mp.KoinPlatformTools.g(SourceFile:12)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.koin.core.instance.SingleInstanceFactory.b(SourceFile:13)04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.koin.core.registry.InstanceRegistry.h(SourceFile:23)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.koin.core.scope.Scope.p(SourceFile:9)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.koin.core.scope.Scope.o(SourceFile:73)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.koin.core.scope.Scope.f(SourceFile:156)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.ligi.passandroid.ui.PassAndroidActivity$special$$inlined$inject$default$3.invoke(SourceFile:17)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at kotlin.SynchronizedLazyImpl.getValue(SourceFile:21)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.ligi.passandroid.ui.PassAndroidActivity.p0(Unknown Source:2)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.ligi.passandroid.ui.PassListActivity.onCreate(SourceFile:67)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:8397)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:8370)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1403)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3841)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        ... 12 more
04-18 18:35:05.759  8663  8663 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: org.ligi.passandroid: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
04-18 18:35:05.759  8663  8663 E AndroidRuntime: Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.PendingIntent.checkFlags(PendingIntent.java:402)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.PendingIntent.getBroadcastAsUser(PendingIntent.java:672)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at android.app.PendingIntent.getBroadcast(PendingIntent.java:659)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at com.google.android.gms.internal.gtm.zzbv.y0(SourceFile:24)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at com.google.android.gms.internal.gtm.zzbv.t0(SourceFile:6)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at com.google.android.gms.internal.gtm.zzbv.r0(SourceFile:1)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at com.google.android.gms.internal.gtm.zzan.q0(SourceFile:1)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at com.google.android.gms.internal.gtm.zzap.<init>(SourceFile:177)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at com.google.android.gms.internal.gtm.zzap.c(SourceFile:30)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at com.google.android.gms.analytics.GoogleAnalytics.k(Unknown Source:0)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.ligi.passandroid.AnalyticsTracker.<init>(SourceFile:6)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.ligi.passandroid.TrackerKodeinModuleKt.a(SourceFile:8)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.ligi.passandroid.App$createKoin$1$3.a(SourceFile:13)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.ligi.passandroid.App$createKoin$1$3.invoke(Unknown Source:4)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        at org.koin.core.instance.InstanceFactory.a(SourceFile:69)
04-18 18:35:05.759  8663  8663 E AndroidRuntime:        ... 29 more
```
Ap Version: 3.7.0
Android Version: 13

**Comment by simontb** [comment]
Same for me

**Comment by ligi** [comment]
Thanks - very strange though - both places where I create PendingIntents I now pass FLAG_IMMUTABLE when SDK level is > 31
Need to fee up some space to create a 31 emulator to reproduce it - thanks for reporting and sorry for the trouble

**Comment by andrewshadura** [comment]
Still crashing every time on my phone.

**Comment by nomis** [comment]
> Thanks - very strange though - both places where I create PendingIntents I now pass FLAG_IMMUTABLE when SDK level is > 31

The stack trace shows that this is caused by Google Analytics, so it's only fixed by 3ecb43596699c6fed9c99d27e6f4926bbfe97b0a in 3.7.1

**Comment by ligi** [comment]
This should be fixed in 3.7.3 which is now in beta of play - thanks for reporting
